test/unit/org/apache/cassandra/db/marshal/LineStringTypeTest.java

```diff
@@ -0,0 +1,150 @@
+    {
+        ByteBuffer actual = type.fromString("linestring(30 10, 10 30, 40 40)");
+
+        ByteBuffer expected = ByteBuffer.allocate(1024).order(ByteOrder.nativeOrder());
+        expected.position(0);
+
+        expected.put((byte) (ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN ? 1 : 0)); // endianness
+        expected.putInt(2);  // type
+        expected.putInt(3);  // num points
+        expected.putDouble(30);  // x1
+        expected.putDouble(10);  // y1
+        expected.putDouble(10);  // x2
+        expected.putDouble(30);  // y2
+        expected.putDouble(40);  // x3
+        expected.putDouble(40);  // y3
+        expected.flip();
+
+        logger.debug("expected: {}", ByteBufferUtil.bytesToHex(expected));
+        logger.debug("actual:   {}", ByteBufferUtil.bytesToHex(actual));
+        String failMsg = String.format("%s != %s", ByteBufferUtil.bytesToHex(actual), ByteBufferUtil.bytesToHex(expected));
+        Assert.assertEquals(failMsg, expected, actual);
+
+        LineString expectedGeometry = lineString(p(30, 10), p(10, 30), p(40, 40));
+        LineString actualGeometry = type.getSerializer().deserialize(actual);
+        logger.debug("expected: {}", expectedGeometry);
+        logger.debug("actual:   {}", actualGeometry);
+        Assert.assertEquals(expectedGeometry, actualGeometry);
+    }
+
+    @Test(expected=MarshalException.class)
+    public void emptyFailure()
+    {
+        type.fromString("linestring()");
+    }
+
+    @Test(expected=MarshalException.class)
+    public void failure3d()
+    {
+        type.fromString("linestring(30 10 20, 10 30 20)");
+    }
+
+    /**
+     * Line strings that cross themselves shouldn't validate
+     */
+    @Test(expected=MarshalException.class)
+    public void simpleFailure()
+    {
+        type.fromString("linestring(0 0, 1 1, 0 1, 1 0)");
+    }
+
+    @Test(expected=MarshalException.class)
+    public void parseFailure()
+    {
+        type.fromString("superlinestring(30 10, 10 30, 40 40)");
+    }
+
+    @Test
+    public void jsonWktInput()
+    {
+        Constants.Value value = (Constants.Value) type.fromJSONObject("linestring(30 10, 10 30, 40 40)");
+        Assert.assertEquals(lineString(p(30, 10), p(10, 30), p(40, 40)), type.getSerializer().deserialize(value.bytes));
+    }
+
+    @Test
+    public void geoJsonInput()
+    {
+        String json = "{\"type\":\"LineString\",\"coordinates\":[[30.0,10.0],[10.0,30.0],[40.0,40.0]]}";
+        Constants.Value value = (Constants.Value) type.fromJSONObject(json);
+        Assert.assertEquals(lineString(p(30, 10), p(10, 30), p(40, 40)), type.getSerializer().deserialize(value.bytes));
+    }
+
+    @Test
+    public void geoJsonOutput()
+    {
+        String json = type.toJSONString(type.getSerializer().serialize(lineString(p(30, 10), p(10, 30), p(40, 40))), ProtocolVersion.CURRENT);
+        Assert.assertEquals("{\"type\":\"LineString\",\"coordinates\":[[30,10],[10,30],[40,40]]}", json);
```

Comment:
In DSE the coordinates are returned as floats with a .0, not sure if this will be a problem that the string is different.
The DSE JSON:
`"{\"type\":\"LineString\",\"coordinates\":[[30.0,10.0],[10.0,30.0],[40.0,40.0]]}"`

Reply:
DSE is able to parse both variants, so is the implementation in dse-db. It should be fine.